### DISCUSSION TITLE
Downloading Unavailable [Private Website]

### DISCUSSION THREAD

**Discussion by iDanishDev** [aclap-dev/video-downloadhelper#513] [Help and Q&A]
### Your question

Trying to download a video from a private source with credientials and on clicking download it fails. Errors are below

### Addon & CoApp version and details

version: 9.0.1.13
target: microsoft
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"E:\\testrec\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36 Edg/124.0.0.0
{
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 4
}

<h3>Bug Copied Below:</h3>

{
  "version": "9.0.1.13",
  "target": "microsoft",
  "channel": "stable",
  "page_url": "https://www.rglectures.com/s/courses/65cdab7ae4b0a1daf0edb960/take",
  "error": {
    "details": "SideDownload error: : [hls @ 000002070883b780] Error when loading first segment 'https://dnqe0v482iyb1.cloudfront.net/spees/w/o/619d5e910cf24fcf26a942ed/v/662453571c32fd3c1fa47c09/u/6641b0957d247c6719e61915/t/6c57b568e0ae9ad5818386668dcd9353/p/assets/videos/619d5e910cf24fcf26a942ed/2024/04/20/662453571c32fd3c1fa47c09/hls_1M_000.ts'\r\n[in#0 @ 000002070883b5c0] Error opening input: Invalid data found when processing input\r\nError opening input file https://dnqe0v482iyb1.cloudfront.net/spees/w/o/619d5e910cf24fcf26a942ed/v/662453571c32fd3c1fa47c09/u/6641b0957d247c6719e61915/t/6c57b568e0ae9ad5818386668dcd9353/p/assets/videos/619d5e910cf24fcf26a942ed/2024/04/20/662453571c32fd3c1fa47c09/hls_1M_.m3u8.\r\nError opening input files: Invalid data found when processing input\r\n",
    "downloadable_id": "downloadable_4350295090229588",
    "error": "coapp_failure",
    "id": "coapp_failure_2d49118f-6531-4fdf-9385-8d6ed53b9cfe"
  }
}

###

![image](https://github.com/aclap-dev/video-downloadhelper/assets/33805784/40af6427-5550-4274-a866-093fec5212d4)

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user, like you.

I'm going to make a wild-ass guess here.  Your antivirus or your firewall or both are blocking your access to this site.  There are several discussions in this forum on that topic.  Search for them & see if your answer is in those.

**Comment by Mattmjs** [comment]
I tried out the m3u8 link to find it hadn't expired in the stable and beta extensions for Firefox. There was a failure on both of them.
I don't think it is a firewall/anti virus issue here. It may be encryption but I'm no expert on any of this. 
It suggests that from the developer tools. The response for the manifest shows only shows a long garbled string.
Normally this would show the master manifest information or a playlist of the segments.
What do you think @paulrouget 

![Screenshot 2024-05-20 145340](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/4f0fbbb8-3c6e-4974-ba96-8b258d23ca27)

**Reply by GrampaWildWilly** [reply]
Your comment got me to look at this again.  I opened the HLS manifest our original poster supplied in his error report (the m3u8 link) in a new tab & dismissed the prompt asking what to do with it.  But I opened it with the Network Monitor active.  That put the m3u8 into the Network Monitor.  All of this was to test whether the URL had timed out, & it seems it has not.  So I pumped the manifest through ffprobe:

[ffprobe (CoApp).txt](https://github.com/aclap-dev/video-downloadhelper/files/15383210/ffprobe.CoApp.txt)

The 502 Bad Gateway error almost sounds like a network issue.  But then it says it couldn't open the key file.  To me, that confirms your suspicion that this is about encryption.  You can see that ffprobe then proceeded to read the first 5 chunks of the stream.  Usually, ffprobe reads only 1 or 2 chunks from the front of the stream.  Reading 5 here indicates some sort of weirdness going on.  The eventual error saying the first chunk got an error when ffprobe tried to load it could very well be a result of the earlier error about inability to read the key file.  I believe the subsequent complaint about invalid data is not truly meaningful.  I believe that error is merely an after effect of the earlier errors & not really informative.  I believe that as a result of failing to read the key file, ffprobe actually didn't read invalid data from the first chunk of the stream.  I believe it failed to read the first chunk at all because it didn't have the encryption key.

The headers you point out that refer to AES256 encryption is an issue we stumbled on back on the Google forum last year.  I found references in . . . I think it was a Reddit discussion . . . that said this had been a problem in ffmpeg several years ago but that it had been resolved by some code change in ffmpeg.  I repeat, this was something that was allegedly corrected a few years ago.  When I get stuff from Medici TV, it is encrypted with a different encryption technique.  I forget what it's called.  I think it has 128 in it.  So this is another encryption technique but ffprobe/ffmpeg appear to not like this one after all.

Now, I am using ffprobe & ffmpeg as supplied by Paul with the CoApp.  You can tell that from the copyright line at the top of the ffprobe output file I attached above.  Maybe there's some encryption code that Paul didn't take from the base ffprobe/ffmpeg.

On the other hand, here's what I get from ffprobe not supplied with the CoApp:

[ffprobe (non CoApp).txt](https://github.com/aclap-dev/video-downloadhelper/files/15383231/ffprobe.non.CoApp.txt)

Note the different copyright notice at the top.  Same results.  That would suggest to me that this isn't about Paul including code in his build of ffprobe.

Or maybe there are parameters you can supply when you build ffprobe/ffmpeg that include additional encryption support that is not normally included in run-of-the-mill builds of the software.

Or maybe I actually have no idea what I'm talking about.  I second your plea to have Paul look in here & see how this all looks to him.

I did save that m3u8 file on my system & look inside it.  It turned out to not be a master manifest but rather a stream manifest.  It shows a reference to a key file:

#EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29

Yet the manifest itself appears to have AES256 encryption.  Maybe that's an intentional mismatch on the part of the web site admins to thwart downloaders.  But I'm just guessing.  I don't really know anything about this stuff.  I would still like to hear what Paul has to say about all this.

**Reply by paulrouget** [reply]
> Maybe there's some encryption code that Paul didn't take from the base ffprobe/ffmpeg

Very possible. I think the HTTP encryption is handled correctly but not the chunk encryption. I'll look.
